### ISSUE TITLE
Recursive Typings

### ISSUE THREAD

**Issue by dschoeni** [ethereumjs/rlp#35]
There seems to be an issue using this lib in Typescript. Consider this example that works perfectly fine in the console:

```
> rlp.encode([[[[[0]]]]])
<Buffer c5 c4 c3 c2 c1 80>
```

Now, writing the same in Typescript won't work:

```
Argument of type 'number[][][][][]' is not assignable to parameter of type 'string | number | Buffer | (string | number | Buffer)[]'.
  Type 'number[][][][][]' is not assignable to type '(string | number | Buffer)[]'.
    Type 'number[][][][]' is not assignable to type 'string | number | Buffer'.
      Type 'number[][][][]' is not assignable to type 'Buffer'.
        Property 'write' is missing in type 'number[][][][]'.
```

I guess there needs to be some way to support recursive types, if that is possible?

**Comment by alcuadrado** [comment]
This issue was fixed in #37. We should close it.

**Comment by holgerd77** [comment]
@alcuadrado Thanks, will close.
